src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Image/ImageParser.cs

```diff
@@ -0,0 +1,289 @@
+                {
+                    logError($"Invalid attribute \"{attributeName}\". Values must be terminated with a double quote.");
+                    return false;
+                }
+                b.Append(c);
+                c = slice.NextChar();
+            }
+            slice.NextChar();
+            value = b.ToString();
+            return true;
+        }
+
+        private bool TryMatchAttributes(ref StringSlice slice, out IDictionary<string, string> attributes, string extensionName, Action<string> logError)
+        {
+            attributes = EmptyAttributes;
+            while (true)
+            {
+                ExtensionsHelper.SkipSpaces(ref slice);
+                if (ExtensionsHelper.MatchStart(ref slice, ":::"))
+                {
+                    return true;
+                }
+                string attributeName;
+                if (!TryMatchIdentifier(ref slice, out attributeName))
+                {
+                    logError($"No attributes exist. source, and alt-text attributes are required. id is optional.");
+                    return false;
+                }
+                if (attributes.ContainsKey(attributeName))
+                {
+                    logError($"Attribute \"{attributeName}\" has been specified multiple times.");
+                    return false;
+                }
+
+                var value = string.Empty;
+
+                ExtensionsHelper.SkipSpaces(ref slice);
+                if (slice.CurrentChar == '=')
+                {
+                    slice.NextChar();
+                    ExtensionsHelper.SkipSpaces(ref slice);
+                    if (!TryMatchAttributeValue(ref slice, out value, extensionName, attributeName, logError))
+                    {
+                        return false;
+                    }
+                }
+
+                if (attributes == EmptyAttributes)
+                {
+                    attributes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                }
+                attributes.Add(attributeName, value);
+            }
+        }
+
+        public bool TryProcessAttributes(IDictionary<string, string> attributes, out HtmlAttributes htmlAttributes, out IDictionary<string, string> renderProperties, Action<string> logWarning)
+        {
+            htmlAttributes = null;
+            renderProperties = new Dictionary<string, string>();
+            var src = string.Empty;
+            var alt = string.Empty;
+            var id = string.Empty;
+            foreach (var attribute in attributes)
+            {
+                var name = attribute.Key;
+                var value = attribute.Value;
+                switch (name)
+                {
+                    case "alt-text":
+                        alt = value;
+                        break;
+                    case "source":
+                        src = value;
+                        break;
+                    case "id":
```

Comment:
Only reason was because It was in the spec that writers should be able to set "id" by themselves, but based on feedback that "id" must be unique, so we are now only allowing it to be generated.